src/client/contexts/CubeContext.tsx

```diff
@@ -831,19 +844,27 @@ export function CubeContextProvider({
       for (const remove of list) {
         // if this card has been edited, remove the edit
         if (newChanges[remove.board]?.edits) {
-          const editIndex = newChanges[remove.board]?.edits?.findIndex((e) => e.index === remove.index) || -1;
-          if (editIndex !== -1) {
-            newChanges[remove.board]?.edits?.splice(editIndex, 1);
-          }
+          removeCardByIndexFromChangeset(newChanges[remove.board]?.edits, remove.index);
+        }
+
+        // if this card has been swapped, remove the swap
+        if (newChanges[remove.board]?.swaps) {
+          removeCardByIndexFromChangeset(newChanges[remove.board]?.swaps, remove.index);
         }
 
         if (!newChanges[remove.board]) {
           newChanges[remove.board] = { adds: [], removes: [], swaps: [], edits: [] };
         }
-        newChanges[remove.board]?.removes?.push({
-          index: remove.index,
-          oldCard: cube.cards[remove.board][remove.index],
-        });
+
+        const removes = newChanges[remove.board]?.removes || [];
+        const removeIndex = removes.findIndex((e) => e.index === remove.index);
+        //Don't add the same card to the removals list
+        if (removeIndex === -1 && removes) {
```

Comment:
Here is the new check that the card being bulk removed isn't already in the removal set